chain/types/ethtypes/eth_types.go

```diff
@@ -815,3 +815,82 @@ func (e EthFeeHistoryParams) MarshalJSON() ([]byte, error) {
 	}
 	return json.Marshal([]interface{}{e.BlkCount, e.NewestBlkNum})
 }
+
+type EthBlockParamByNumberOrHash struct {
+	PredefinedBlock *string
+	Number          *EthUint64
```

Comment:
@Stebalien I looked better into merging the `Number`/`BlockNumber` into one and I think it may not be such a great idea.

For context, the issue is that currently the `blkParam` can be one of "latest", "earliest", "pending", or a block number (decimal or hex) but with this PR we additionally need to support that `blkParam` to be able to be a json hash witch a certain named fields (blockNumber,blockHash,requireCanonical).  To implement this, I introduced this `EthBlockParamByNumberOrHash` struct which has fields for both cases (normal as before + support for json hashes) and handles the al the construction/json deserialization. 

It would be possible to merge the `Number` and `BlockNumber` in `EthBlockParamByNumberOrHash`into one, but I would think that makes the code less readable since right now its easy to see from reading the code whether we are talking about a normal block number or a one we need to support for the EIP-1891. Also, it would make error handling cumbersome (like [here](https://github.com/filecoin-project/lotus/pull/10815/files#diff-b4b04abadd60724f3b7710cced91031f331c2edacb30a1c1bb7d5aeb67fc414cR103-R106)).